client/my-sites/hosting/github/use-github-connection.ts

```diff
@@ -0,0 +1,40 @@
+import { useQuery, UseQueryOptions } from 'react-query';
+import wp from 'calypso/lib/wp';
+
+export const USE_GITHUB_REPOS_QUERY_KEY = 'github-connection-query-key';
+
+export type InactiveGitHubConnection = {
+	connected: boolean;
+};
+
+export type GitHubConnection = {
+	ID: number;
+	base_path: string;
+	repo: string;
+	branch: string;
+	external_profile_picture: string;
+	external_name: string;
+	label: string;
+} & InactiveGitHubConnection;
```

Comment:
You can model this sort of data type with a discriminated union.

```suggestion
export interface InactiveGitHubConnection {
	connected: false;
}

export interface ActiveGitHubConnection {
	connected: true;
	ID: number;
	base_path: string;
	repo: string;
	branch: string;
	external_profile_picture: string;
	external_name: string;
	label: string;
}

// This is the union in "discriminated union"
export type GitHubConnection = InactiveGitHubConnection | ActiveGitHubConnection;
```

Doing it this way means that after you check the value of `connected`, TypeScript knows which of the two types it is. So you should also change the prop type of `GithubConnectCardProps.connection` to `ActiveGitHubConnection`.